vars/dockerExecute.groovy

```diff
@@ -1,51 +1,65 @@
 import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.ConfigurationLoader
 
 def call(Map parameters = [:], body) {
 
     def STEP_NAME = 'dockerExecute'
     def PLUGIN_ID_DOCKER_WORKFLOW = 'docker-workflow'
 
-    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters){
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
         def dockerImage = parameters.dockerImage ?: ''
         Map dockerEnvVars = parameters.dockerEnvVars ?: [:]
         def dockerOptions = parameters.dockerOptions ?: ''
         Map dockerVolumeBind = parameters.dockerVolumeBind ?: [:]
+        final script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+        if (env.POD_NAME && hasContainerDefined(script, dockerImage)) {
+            container(getContainerDefined(script, dockerImage)) {
+                echo "Executing inside a Kubernetes Container"
+                body()
+                sh "chown -R 1000:1000 ."
+            }
+        } else if (env.jaas_owner) {
+            executeDockerOnKubernetes(
+                dockerImage: parameters.dockerImage,
+                dockerEnvVars: parameters.dockerEnvVars,
+                dockerOptions: parameters.dockerOptions,
+                dockerVolumeBind: parameters.dockerVolumeBind) {
```

Comment:
Currently not possible within JaaS environment

Reply:
This is OS Piper. No SAP specifics, internals etc allowed here.